src/models/krate_owner_actions.rs

```diff
@@ -0,0 +1,101 @@
+use chrono::NaiveDateTime;
+use diesel::prelude::*;
+use diesel::{
+    deserialize::{self, FromSql},
+    pg::Pg,
+    serialize::{self, Output, ToSql},
+    sql_types::Integer,
+};
+use std::io::Write;
+
+use crate::models::{ApiToken, Crate, User};
+use crate::schema::*;
+
+#[derive(Debug, Clone, Copy, PartialEq, FromSqlRow, AsExpression)]
+#[repr(i32)]
+#[sql_type = "Integer"]
+pub enum CrateAction {
```

Comment:
I'd prefer a name like `ActionKind` or `ActionType` here, or even `CrateOwnerActionKind`, though this is kind of unwieldy.

We have the `CrateOwnerAction` and `CrateAction` types in this module, and while these types have quite different nature, the nature of their difference isn't reflected in the current naming.

Reply:
I like the suggestion. I'll have a look.